registry/handlers/tags.go

```diff
@@ -49,6 +51,39 @@ func (th *tagsHandler) GetTags(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// do pagination if requested
+	q := r.URL.Query()
+	// get entries after latest, if any specified
+	if lastEntry := q.Get("last"); lastEntry != "" {
+		lastEntryIndex := sort.SearchStrings(tags, lastEntry)
+		if lastEntryIndex != len(tags) && tags[lastEntryIndex] == lastEntry {
+			tags = tags[lastEntryIndex+1:]
+		}
+	}
+
+	// if no error, means that the user requested `n` entries
+	if maxEntries, err := strconv.Atoi(q.Get("n")); err == nil {
```

Comment:
I think it makes more sense here to return 400 if n is a garbage value rather than just ignore it. Just go into this block if `n` is set.

Reply:
Sure, I can do that! 
Keep in mind that atm. catalog has the same behavior, where it will not throw any error on a invalid `n` value (https://github.com/docker/distribution/blob/master/registry/handlers/catalog.go#L41-L44).

Should I make a PR to get them to match up?

Reply:
I think that is the right behavior, but considering that API is not used by end clients, it's not super important.

We can possibly discuss the error handling of these in the distribution spec as well if it is too ambiguous. Handling bad input (n is not a number, last does not exist) could be clearly defined.

Reply:
Lets get this merged first then I can get the other branch up-to-date and use the distribution spec in that as well.
Somewhat easy to do for `n`, but when `last` is wrong it is a different case (either the tag does not exists or the repository doesn't).

I currently have this wording for the spec, but feel unsure about the wording.

```go
	// ErrorPaginationNumberInvalid is returned when the requested number
	// of results n is not a integer.
	ErrorPaginationNumberInvalid = errcode.Register(errGroup, errcode.ErrorDescriptor{
		Value:   "PAGINATION_NUMBER_INVALID",
		Message: "invalid number of results requested",
		Description: `This error may be returned when the requested
		"n" number of results is not an integer type`,
		HTTPStatusCode: http.StatusBadRequest,
	})
```

Would love some input, and maybe a suggestion for the `last` param. Else I will try to word it myself tomorrow :)

Reply:
@dmcgowan fixed in latest commits, please check to ensure it is correct :)

Reply:
Hoping a ping is OK,new Github notifications isn't nice @dmcgowan :|

Reply:
I saw this, but wanted to cross reference these changes with the spec first before merging. Just went through now, the section is https://github.com/opencontainers/distribution-spec/blob/master/spec.md#tags-paginated